drivers/spi/spi_ambiq_spis.c

```diff
@@ -0,0 +1,381 @@
+
+	data->ios_cfg.ui32InterfaceSelect = AM_HAL_IOS_USE_SPI;
+
+	if (spi_context_configured(ctx, config)) {
+		/* Already configured. No need to do it again. */
+		return 0;
+	}
+
+	if (SPI_WORD_SIZE_GET(config->operation) != SPI_WORD_SIZE) {
+		LOG_ERR("Word size must be %d", SPI_WORD_SIZE);
+		return -ENOTSUP;
+	}
+
+	if ((config->operation & SPI_LINES_MASK) != SPI_LINES_SINGLE) {
+		LOG_ERR("Only supports single mode");
+		return -ENOTSUP;
+	}
+
+	if (config->operation & SPI_LOCK_ON) {
+		LOG_ERR("Lock On not supported");
+		return -ENOTSUP;
+	}
+
+	if (config->operation & SPI_TRANSFER_LSB) {
+		LOG_ERR("LSB first not supported");
+		return -ENOTSUP;
+	}
+
+	if (config->operation & SPI_MODE_CPOL) {
+		if (config->operation & SPI_MODE_CPHA) {
+			data->ios_cfg.ui32InterfaceSelect |= AM_HAL_IOS_SPIMODE_3;
+		} else {
+			data->ios_cfg.ui32InterfaceSelect |= AM_HAL_IOS_SPIMODE_2;
+		}
+	} else {
+		if (config->operation & SPI_MODE_CPHA) {
+			data->ios_cfg.ui32InterfaceSelect |= AM_HAL_IOS_SPIMODE_1;
+		} else {
+			data->ios_cfg.ui32InterfaceSelect |= AM_HAL_IOS_SPIMODE_0;
+		}
+	}
+
+	if (config->operation & SPI_OP_MODE_MASTER) {
+		LOG_ERR("Controller mode not supported");
+		return -ENOTSUP;
+	}
+	if (config->operation & SPI_MODE_LOOP) {
+		LOG_ERR("Loopback mode not supported");
+		return -ENOTSUP;
+	}
+
+	if (spi_cs_is_gpio(config)) {
+		LOG_ERR("CS control via GPIO is not supported");
+		return -EINVAL;
+	}
+
+	/* Eliminate the "read-only" section, so an external host can use the
+	 * entire "direct write" section.
+	 */
+	data->ios_cfg.ui32ROBase = FIFO_BASE;
+	/* Making the "FIFO" section as big as possible. */
+	data->ios_cfg.ui32FIFOBase = FIFO_BASE;
+	/* We don't need any RAM space, so extend the FIFO all the way to the end
+	 * of the LRAM.
+	 */
+	data->ios_cfg.ui32RAMBase = FIFO_END;
+	/* FIFO Threshold - set to half the size */
+	data->ios_cfg.ui32FIFOThreshold = FIFO_LENGTH >> 1;
+
+	data->ios_cfg.pui8SRAMBuffer = g_pui8TxFifoBuffer,
+	data->ios_cfg.ui32SRAMBufferCap = AMBIQ_SPIS_TX_BUFSIZE_MAX,
+
+	ctx->config = config;
+
+	ret = am_hal_ios_configure(data->ios_handler, &data->ios_cfg);
+
+	return ret;
+}
+
+static int spi_ambiq_xfer(const struct device *dev, const struct spi_config *config)
+{
+	const struct spi_ambiq_config *cfg = dev->config;
+	struct spi_ambiq_data *data = dev->data;
+	struct spi_context *ctx = &data->ctx;
+	int ret = 0;
+	uint32_t chunk, num_written, num_read = 0;
+
+	while (1) {
+		if (spi_context_tx_buf_on(ctx)) {
+			/* Inform the host */
+			gpio_pin_set_dt(&cfg->int_gpios, 1);
+			k_sleep(K_USEC(100));
+			gpio_pin_set_dt(&cfg->int_gpios, 0);
+			/*  Copy data into FIFO */
+			chunk = (ctx->tx_len > AMBIQ_SPIS_TX_BUFSIZE_MAX)
+					? AMBIQ_SPIS_TX_BUFSIZE_MAX
+					: ctx->tx_len;
+			ret = am_hal_ios_fifo_write(data->ios_handler, (uint8_t *)ctx->tx_buf,
+						    chunk, &num_written);
+			if (ret != 0) {
+				spi_ambiq_reset(dev);
+				LOG_ERR("SPIS write error: %d", ret);
+				return ret;
```

Comment:
should be a goto error;

Reply:
@tbursztyka SHOULD or COULD. I dm not aware any documentation in Zephyr that requires the use of goto error statements.